apps/client/src/features/app-settings/panel/general-panel/GeneralPanelForm.tsx

```diff
@@ -0,0 +1,145 @@
+          error={errors.serverPort?.message}
+        >
+          <Input
+            id='serverPort'
+            size='sm'
+            type='number'
+            variant='ontime-filled'
+            maxLength={5}
+            width='75px'
+            {...register('serverPort', {
+              required: { value: true, message: 'Required field' },
+              max: { value: 65535, message: 'Port must be within range 1024 - 65535' },
+              min: { value: 1024, message: 'Port must be within range 1024 - 65535' },
+              pattern: {
+                value: isOnlyNumbers,
+                message: 'Value should be numeric',
+              },
+            })}
+          />
+        </GeneralSplitInput>
+        <GeneralSplitInput
+          field='editorKey'
+          title='Editor pin code'
+          description='Protect the editor with a pin code'
+          error={errors.editorKey?.message}
+        >
+          <GeneralPinInput register={register as any} formName='editorKey' isDisabled={disableInputs} />
+        </GeneralSplitInput>
+        <GeneralSplitInput
+          field='operatorKey'
+          title='Operator pin code'
+          description='Protect the cuesheet with a pin code'
+          error={errors.operatorKey?.message}
+        >
+          <GeneralPinInput register={register as any} formName='operatorKey' isDisabled={disableInputs} />
+        </GeneralSplitInput>
+        <GeneralSplitInput
+          field='timeFormat'
+          title='Time Format'
+          description='Views 12 / 24 hours'
+          error={errors.timeFormat?.message}
+        >
+          <Select size='sm' width='auto' isDisabled={disableInputs} {...register('timeFormat')}>
+            <option value='12'>12 hours eg. 11:00:10 PM</option>
+            <option value='24'>24 hours eg. 23:00:10</option>
+          </Select>
+        </GeneralSplitInput>
+        <GeneralSplitInput
+          field='language'
+          title='Views Language'
+          description='Language for static fields in views'
+          error={errors.language?.message}
+        >
+          <Select size='sm' width='auto' isDisabled={disableInputs} {...register('language')}>
+            <option value='en'>English</option>
+            <option value='fr'>French</option>
+            <option value='de'>German</option>
+            <option value='it'>Italian</option>
+            <option value='no'>Norwegian</option>
+            <option value='pt'>Portuguese</option>
+            <option value='es'>Spanish</option>
+            <option value='sv'>Swedish</option>
+          </Select>
+        </GeneralSplitInput>
+        <GeneralFooter
+          formId='app-settings'
+          handleRevert={onReset}
+          isDirty={isDirty}
+          isValid={isValid}
+          isSubmitting={isSubmitting}
+        />
+      </form>
+      {submitError && <span className={style.error}>{submitError}</span>}
```

Comment:
hi, thank you for your work,
I think this generally looks fine, we just need a few tweaks to UI that I can help with.

Meanwhile, the old structure of the components is a bit cumbersome as you found with having to pass the references. In the new version of settings I have opted from leaving this at top level.

I hope this makes it easier to develop. I feel that our app settings are not particularly complex and dont warrant this abstraction

have a look here for how we could do it
https://github.com/cpvalente/ontime/blob/v3/apps/client/src/features/app-settings/panel/integrations-panel/OscIntegrations.tsx#L102

Reach out if this is unclear